docs/spec.md

```diff
@@ -2633,6 +2633,54 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills a `result` tensor with values in increasing order starting from zero
```

Comment:
Fix link in StablehloOps.td to point at dynamic_iota:

`https://github.com/openxla/stablehlo/blob/main/docs/spec.md#iota`

Reply:
```
Check that the description in the op's ODS:
Includes the first sentence of the spec.
Then links to the corresponding section of the spec.
Then uses the same example as the spec but via pretty syntax which can be obtaining by running stablehlo-opt.
```